### ISSUE TITLE
DJ Light: not as good as in SR 0.13.3

### ISSUE THREAD

**Issue by softhack007** [MoonModules/WLED#12]
### What happened?

From Artacus @ discord — today 17:03
On MM fork, I’d always get green dot at top left pixel.  On AC, it was barely reacting to sound and all the bars in a given frame would be the same color. So while it looked like it it was working, when it was side by side with a 0.13 render, it looked terrible in comparison.

### To Reproduce Bug

see above

This is a "reminder to myself" -> need to review the source code and compare it to SR 0.13.3 code

### Expected Behavior

looks as awesome as always :-)

### Install Method

Self-Compiled or other

### What version/release of MM WLED?

WLED main development branch

### Which microcontroller/board are you seeing the problem on?

ESP32

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by softhack007** [comment]
Two differences between 0.14.0 and SR 0.13.x could be related:
* Sampling frequency in 0.14. is 22kHz, while in 0.13 it was 10Khz. So GEQ channels represent other frequencies.
* SR 0.13.3 does not have the new code to "balance out" GEQ channels.  0.14 amplifies higher channels, to get an "equal" response in all channels. 
So in 0.13 the lower channels are a lot stronger than others, and "blue" usually wins. In 0.14 the channels are balanced, this could lead to having more "pastel" colors in DJLight effect.

**Comment by MoonModules** [comment]
@softhack007 should we close this?

**Comment by softhack007** [comment]
@ewoudwijma, @netmindz : The problem is due to two differences in 0.14:
- fftResult channels contain different frequencies
- fftResult are equalized (according mic profiles + frequency scaling) in 0.14, while in 0.13 they were always having the strongest peaks at low frequencies. You can see the difference if comparing GEQ in 0.13 vs. 0.14,

@netmindz: would be great if you can tinker with the DJ Light effect a bit more, I think my results are not optimal. PR welcome, as always :-)

**Comment by netmindz** [comment]
See you've made some tweaks already @softhack007 , were you aiming to match the original output out just to give an "update" to make look pretty again?

**Comment by softhack007** [comment]
> were you aiming to match the original output out just to give an "update" to make look pretty again?

Both actually.
First i tried to just restore the original look, by changing frequency bands according to the new mapping in 0.14.
That still gave me too much white and pastels, so I've tried to go RGB->HSV, then change saturation, then back to RGB. This looks better, but still could need some tweaking of parameters.

Finally I recognized the original code worked because the frequencies in 0.13 were not "equalized" as they are in 0.14. So I tried some other frequency-to-color mapping. That's still experimental, as I recognized i don't know enough about how the original effect was expected to look.
